File: Python3/Examples/factorial.py
"""
Author: Medic5700

This is an example of generating a factorial using recursion.
"""

def factorial (n : int) -> int:
    """Takes in a number n, returns the factorial of n using recursion"""
    if (n > 1):
        return n * factorial (n - 1)
    else:
        return 1

if __name__ == "__main__":
    for i in range(0,11):
        print(str(i) + "!\t= " + str(factorial(i)))

File: C/Examples/factorial.c
/**
 * Authour: Medic5700
 * 
 * Print factorials
**/

#include<stdio.h>

int factorial(int t1){
	if (t1 > 1){
		return t1 * factorial(t1 - 1);
	}
	else{
		return 1;
	}
}
int main(){
	int i;
	for (i=0; i<=10; i++){
		printf("%d!\t= %d\n", i, factorial(i));
	}
	return 0;
}
